js/apps/system/_admin/aardvark/APP/react/src/views/collections/indices/listIndices/CollectionIndexActionButtons.tsx

```diff
@@ -0,0 +1,62 @@
+import { DeleteIcon, LockIcon } from "@chakra-ui/icons";
+import { Box, IconButton, Tooltip } from "@chakra-ui/react";
+import { Index } from "arangojs/indexes";
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { encodeHelper } from "../../../../utils/encodeHelper";
+import { useCollectionIndicesContext } from "../CollectionIndicesContext";
+import { DeleteIndexModal } from "../DeleteIndexModal";
+
+export function CollectionIndexActionButtons({ index }: { index: Index }) {
+  const { readOnly } = useCollectionIndicesContext();
+  const [showDeleteModal, setShowDeleteModal] = React.useState(false);
+  const collectionName = index.id.slice(0, index.id.indexOf("/"));
+  const { encoded: encodedCollectionName } = encodeHelper(collectionName);
+  const history = useHistory();
+
+  const isSystem =
+    index.type === "primary" ||
+    /**
+     * @todo
+     * typecast required because arangojs
+     * doesn't support "edge" index types yet
+     */
+
+    (index.type as string) === "edge";
```

Comment:
So the idea is to make TS look at it that way or actually cast it? Curios..

Reply:
`index.type` can't be `edge` according to arangoJS, so this typecasts it to a string

without this, TS complains that the condition will always be false